sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -624,17 +627,26 @@ private ProxyOptions getDefaultProxyConfiguration(Configuration configuration) {
             return ProxyOptions.SYSTEM_DEFAULTS;
         }
 
-        final String[] hostPort = proxyAddress.split(":");
-        if (hostPort.length < 2) {
-            throw logger.logExceptionAsError(new IllegalArgumentException("HTTP_PROXY cannot be parsed into a proxy"));
-        }
-
-        final String host = hostPort[0];
-        final int port = Integer.parseInt(hostPort[1]);
-        final Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(host, port));
-        final String username = configuration.get(ProxyOptions.PROXY_USERNAME);
-        final String password = configuration.get(ProxyOptions.PROXY_PASSWORD);
+        return getProxyOptions(authentication, proxyAddress);
+    }
 
-        return new ProxyOptions(authentication, proxy, username, password);
+    private ProxyOptions getProxyOptions(ProxyAuthenticationType authentication, String proxyAddress) {
+        String host;
+        int port;
+        if (HOST_PORT_PATTERN.matcher(proxyAddress.trim()).find()) {
```

Comment:
This logic doesn't exist in SB either.... would you mind copy and paste it there? :D Kill two birds with 3 stones?

Reply:
Sure, let me fix SB too.

Reply:
You're the best!
![](https://media2.giphy.com/media/26AHAw0aMmWwRI4Hm/giphy.gif)